pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -37,24 +41,27 @@ type PeerFinder interface {
 }
 
 type Service struct {
-	peerFinder PeerFinder
-	overlay    swarm.Address
-	networkID  int32
-	logger     logging.Logger
+	overlay            swarm.Address
+	networkID          int32
+	receivedHandshakes map[libp2ppeer.ID]struct{}
+	mtx                sync.Mutex // guards received handshakes
```

Comment:
comment not necessary. you can also rename the mutex to be `muReceivedHandshakes`

Reply:
Yes that would make sense. I would prefer `receivedHandshakesMu` name.